tests/test_schemas.py

```diff
@@ -613,3 +616,95 @@ def post(self, request, *args, **kwargs):
     descr = schema.get_description('example', 'get')
     # the first and last character are '\n' correctly removed by get_description
     assert descr == formatting.dedent(ExampleDocstringAPIView.__doc__[1:][:-1])
+
+
+# Views for SchemaGenerationExclusionTests
+class ExcludedAPIView(APIView):
+    schema = None
+
+    def get(self, request, *args, **kwargs):
+        pass
+
+
+@api_view(['GET'])
+@schema(None)
+def excluded_fbv(request):
+    pass
+
+
+@api_view(['GET'])
+def included_fbv(request):
+    pass
+
+
+@unittest.skipUnless(coreapi, 'coreapi is not installed')
+class SchemaGenerationExclusionTests(TestCase):
+    def setUp(self):
+        self.patterns = [
+            url('^excluded-cbv/$', ExcludedAPIView.as_view()),
+            url('^excluded-fbv/$', excluded_fbv),
+            url('^included-fbv/$', included_fbv),
+        ]
+
+    def test_schema_generator_excludes_correctly(self):
+        """Schema should not include excluded views"""
+        generator = SchemaGenerator(title='Exclusions', patterns=self.patterns)
+        schema = generator.get_schema()
+        expected = coreapi.Document(
+            url='',
+            title='Exclusions',
+            content={
+                'included-fbv': {
+                    'list': coreapi.Link(url='/included-fbv/', action='get')
+                }
+            }
+        )
+
+        assert len(schema.data) == 1
+        assert 'included-fbv' in schema.data
+        assert schema == expected
+
+    def test_endpoint_enumerator_excludes_correctly(self):
+        """It is responsibility of EndpointEnumerator to exclude views"""
+        inspector = EndpointEnumerator(self.patterns)
+        endpoints = inspector.get_api_endpoints()
+
+        assert len(endpoints) == 1
+        path, method, callback = endpoints[0]
+        assert path == '/included-fbv/'
+
+    def test_should_include_endpoint_excludes_correctly(self):
+        """This is the specific method that should handle the exclusion"""
+        inspector = EndpointEnumerator(self.patterns)
+
+        # Not pretty. Mimics internals of EndpointEnumerator to put should_include_endpoint under test
+        pairs = [(inspector.get_path_from_regex(pattern.regex.pattern), pattern.callback)
+                 for pattern in self.patterns]
+
+        should_include = [
+            inspector.should_include_endpoint(*pair) for pair in pairs
+        ]
+
+        expected = [False, False, True]
+
+        assert should_include == expected
+
+    def test_deprecations(self):
```

Comment:
@rpkilby I'm getting console noise in the test run from this. (Well from the call site but...) 

```
tests/test_schemas.py ............/Users/carlton/Documents/Django-Stack/django-rest-framework/rest_framework/decorators.py:83: PendingDeprecationWarning: The `exclude_from_schema` argument to `api_view` is deprecated. Use the `schema` decorator instead, passing `None`.
  PendingDeprecationWarning
/Users/carlton/Documents/Django-Stack/django-rest-framework/rest_framework/schemas/generators.py:156: PendingDeprecationWarning: OldFashjonedExcludedView. The `APIView.exclude_from_schema` is deprecated. Set `schema = None` instead
  warnings.warn(msg, PendingDeprecationWarning)
....
tests/test_serializer.py .........................................
```

Can you advise on the best way to keep that quiet? Ta!

Reply:
Not sure we need to test this. Don't believe that we've done so when deprecating other bits of API.

Reply:
https://github.com/encode/django-rest-framework/blob/efff9ff338bb0994b3249fa9907bb7018d7b6d5e/rest_framework/filters.py#L48-L52

It's something @rpkilby always puts in. I thought, Why not? 🙂

Reply:
It looks like `pytest.warns` isn't recommended for deprecation warnings, as it doesn't capture output.

https://docs.pytest.org/en/latest/warnings.html#ensuring-function-triggers

Reply:
@rpkilby OK. Thanks. I read that too but was still getting the noise... 

1. I went with `pytest.warns(PendingDeprecationWarning)` because (for me) `pytest.deprecated_call()` failed with `DID NOT WARN`. No ideas why that wasn't working. 
2.  Despite the noise locally for me, [this is not appearing in the travis run](https://travis-ci.org/encode/django-rest-framework/jobs/275519390#L517-L520) — so it must be my local setup. 

(There is plenty of noise from py.test in travis but I assume that'll disappear with time.) 

Lets call this done. Thanks for the input! 


@tomchristie I think we're good to go.

Reply:
> Despite the noise locally for me, this is not appearing in the travis run — so it must be my local setup.

No noise when running the tests on my machine, so that would make sense.